extension/js/add_input_icon.js

```diff
@@ -1,33 +1,98 @@
-(function () {
-  for (const emailInput of document.querySelectorAll("input[type='email']")) {
-    const divEl = document.createElement("div");
-    divEl.classList.add("relay-icon");
-
-    const buttonEl = document.createElement("button");
-    buttonEl.id = "relay-button";
-    buttonEl.classList.add("relay-button");
-    buttonEl.type = "button";
-
-    const imgEl = document.createElement("img");
-    imgEl.src = browser.runtime.getURL("icons/placeholder-logo-beta.png");
-    imgEl.classList.add("relay-icon-img");
-    buttonEl.appendChild(imgEl);
-
-    buttonEl.addEventListener("click", async (e) => {
-      const newRelayAddressResponse = await browser.runtime.sendMessage({
-          method: "makeRelayAddress",
-        });
-
-        if (newRelayAddressResponse.status === 402) {
-          emailInput.value = "Too many email aliases.";
-          return;
-        }
-
-        navigator.clipboard.writeText(newRelayAddressResponse);
-        emailInput.value = newRelayAddressResponse;
+function addRelayIconToInput(emailInput) {
+  // remember the input's original parent element;
+  const emailInputOriginalParentEl = emailInput.parentElement;
+
+  // create new wrapping element;
+  const emailInputWrapper = document.createElement("div");
+  emailInputWrapper.classList.add("relay-email-input-wrapper");
+
+  const inputClone = emailInput.cloneNode();
+  // add padding to the cloned input so that input text
+  // is not covered up by the Relay icon
+  inputClone.style.paddingRight = "50px !important;"
+
+
+  emailInputWrapper.appendChild(inputClone);
+
+  const inputHeight = emailInput.clientHeight;
+
+  const divEl = document.createElement("div");
+  divEl.classList.add("relay-icon");
+  divEl.style.height = inputHeight+"px";
+
+  const buttonEl = document.createElement("button");
+  buttonEl.id = "relay-button";
+  buttonEl.classList.add("relay-button");
+  buttonEl.type = "button";
+  buttonEl.title = "Generate new email alias";
+
+  const imgEl = document.createElement("img");
+  imgEl.src = browser.runtime.getURL("icons/make-new-alias.png");
+  imgEl.classList.add("relay-icon-img");
+  buttonEl.appendChild(imgEl);
+
+  const relayIconHeight = 30;
+  if (relayIconHeight > inputHeight) {
+    const smallIconSize = "20px";
+    imgEl.style.height = smallIconSize;
+    imgEl.style.width = smallIconSize;
+    imgEl.style.minWidth = smallIconSize;
+    imgEl.style.minHeight = smallIconSize;
+    inputClone.style.paddingRight = "30px";
+    divEl.style.right = "2px";
+  }
```

Comment:
Similar concern here - all these `.style.* =` directives might break on a site that has CSP that disallows inline styles.

Reply:
Same comment as above. I don't _think_ Monitor allows unsafe-inline (although looking through our CSP policy there, I don't see unsafe-inline explicitly disallowed) yet I was able to get away with this.